providers/env-var/src/main/java/dev/openfeature/contrib/providers/envvar/EnvironmentKeyTransformer.java

```diff
@@ -0,0 +1,113 @@
+ * a transforming function in the constructor. <br>
+ * Currently the following transformations are supported out of the box:
+ * <ul>
+ *     <li>{@link #toLowerCaseTransformer() converting to lower case}</li>
+ *     <li>{@link #toUpperCaseTransformer() converting to UPPER CASE}</li>
+ *     <li>{@link #hyphenCaseToScreamingSnake() converting hyphen-case to SCREAMING_SNAKE_CASE}</li>
+ *     <li>{@link #toCamelCaseTransformer() convert to camelCase}</li>
+ *     <li>{@link #replaceUnderscoreWithDotTransformer() replace '_' with '.'}</li>
+ *     <li>{@link #replaceDotWithUnderscoreTransformer() replace '.' with '_'}</li>
+ * </ul>
+ *
+ * <p><strong>Examples:</strong>
+ *
+ * <p>1. hyphen-case feature flag names to screaming snake-case environment variables:
+ * <pre>
+ * {@code
+ * // Definition of the EnvVarProvider:
+ * EnvironmentGateway transformingGateway = EnvironmentKeyTransformer
+ *     .hyphenCaseToScreamingSnake()
+ *     .withSource(new OS());
+ *
+ * FeatureProvider provider = new EnvVarProvider(transformingGateway);
+ * }
+ * </pre>
+ * 2. chained/composed transformations:
+ * <pre>
+ * {@code
+ * // Definition of the EnvVarProvider:
+ * EnvironmentGateway transformingGateway = EnvironmentKeyTransformer
+ *     .toLowerCaseTransformer()
+ *     .andThen(EnvironmentKeyTransformer.replaceUnderscoreWithDotTransformer())
+ *     .withSource(actualGateway);
+ *
+ * FeatureProvider provider = new EnvVarProvider(transformingGateway);
+ * }
+ * </pre>
+ * 3. freely defined transformation function:
+ * <pre>
+ * {@code
+ *
+ * // Definition of the EnvVarProvider:
+ * EnvironmentGateway transformingGateway = new EnvironmentKeyTransformer(key -> key.substring(1))
+ *     .withSource(actualGateway);
+ *
+ * FeatureProvider provider = new EnvVarProvider(transformingGateway);
+ * }
+ * </pre>
+ */
+@RequiredArgsConstructor
+public class EnvironmentKeyTransformer implements EnvironmentGateway {
+
+    private static final UnaryOperator<String> TO_LOWER_CASE = StringUtils::lowerCase;
+    private static final UnaryOperator<String> TO_UPPER_CASE = StringUtils::upperCase;
+    private static final UnaryOperator<String> TO_CAMEL_CASE = s -> CaseUtils.toCamelCase(s, false, '_');
+    private static final UnaryOperator<String> REPLACE_DOT_WITH_UNDERSCORE = s -> StringUtils.replaceChars(s, ".", "_");
+    private static final UnaryOperator<String> REPLACE_UNDERSCORE_WITH_DOT = s -> StringUtils.replaceChars(s, "_", ".");
+    private static final UnaryOperator<String> REPLACE_HYPHEN_WITH_UNDERSCORE =
+        s -> StringUtils.replaceChars(s, "-", "_");
+
+    private final Function<String, String> transformation;
+
+    @Override
+    public String getEnvironmentVariable(String key) {
+        return transformation.apply(key);
+    }
+
+    public EnvironmentKeyTransformer andThen(EnvironmentGateway another) {
+        return new EnvironmentKeyTransformer(transformation.compose(another::getEnvironmentVariable));
```

Comment:
Minor - assuming the intention here is to first apply this transformation and then apply the `another`, we can use `andThen` instead of `compose` [1]. But let me know if the idea is different here :) 

[1] - https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#andThen-java.util.function.Function-

Reply:
i think, looking at the context and the names of the methods i chose, `andThen` makes more sense 😄 . thx for pointing it out.